pywbem/_nocaselist.py

```diff
@@ -0,0 +1,326 @@
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+
+"""
+Class ``NocaseList`` is a list implementation with case-insensitive but
+case-preserving items.
+
+Whenever items of the list are looked up by value or item values are
+compared, that is done case-insensitively. Whenever the value of list items
+is returned, they have the lexical case that was originally specified.
+
+It is used for lists of case-insensitive items such as namespaces returned
+by the pywbem mock support.
+
+Users of pywbem will notice ``NocaseList`` objects only as a result of pywbem
+functions. Users cannot create ``NocaseList`` objects.
+
+Except for the case-insensitivity of its items, it behaves like the built-in
```

Comment:
Is there anything that can be done with a list that cannot be done with the no case list? Ex. iteration/

Reply:
No.